ga2vcf.py

```diff
@@ -0,0 +1,91 @@
+"""
+Conversion utility that fetches data from the GA4GH server and
+writes to VCF files
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import pysam
+
+import ga4gh.client as client
+import ga4gh.protocol as protocol
+
+
+class VCF(object):
+    """
+    A convenience wrapper around pysam.VCF functionality
+    providing methods to write a VCF file.
+    """
+    
+    infoLineFormatStr = '<ID={id},Number={number},Type={type},Description="{description}"' 
+
+    def __init__(self, filename, variantSet):
+        self.vcf = pysam.VCF()
+        self.stream = file(filename, 'w')
+        self.variantSet = variantSet
+        self.data = []
+
+    def writeHeader(self):
+        header = self._formatVariantSetToVcfHeader()
+        self.vcf.setheader(header)
+        self.vcf.write(self.stream, self.data)
+
+    def close(self):
+        self.stream.close()
+
+    def _formatVariantSetToVcfHeader(self):
+        header = []
+        for metadata in self.variantSet.metadata:
+            infoLine = self._formatMetadataToMetainfoLine(metadata)
+            header.append(infoLine)
+        return header
+
+    def _formatMetadataToMetainfoLine(self, metadata):
+        key = "INFO"
```

Comment:
At first glance:

   1. Better to create VCF 4.1 or 4.2 files.  The 4.0 spec is very dated.
   2. types are not right.  Based on the spec, they are case sensitive and
   must be one of: Integer, Float, Character, String or Flag (for INFO only)
   3. "Number=0" is nonsensical. If Number is 0 then the Type should be
   Flag. If Number is not known, then it should be ".".

I'll look again and more deeply in a bit.

On Wed, Feb 4, 2015 at 11:14 AM, Danny Colligan <notifications@github.com>
wrote:

> In ga2vcf.py
> <https://github.com/ga4gh/server/pull/126#discussion_r24110098>:
>
> > +        header = self._formatVariantSetToVcfHeader()
> > +        self.vcf.setheader(header)
> > +        self.vcf.write(self.stream, self.data)
> > +
> > +    def close(self):
> > +        self.stream.close()
> > +
> > +    def _formatVariantSetToVcfHeader(self):
> > +        header = []
> > +        for metadata in self.variantSet.metadata:
> > +            infoLine = self._formatMetadataToMetainfoLine(metadata)
> > +            header.append(infoLine)
> > +        return header
> > +
> > +    def _formatMetadataToMetainfoLine(self, metadata):
> > +        key = "INFO"
>
> How to distinguish INFO metainfo line from others? (e.g. FORMAT, etc.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ga4gh/server/pull/126/files#r24110098>.
>